File: integration-tests/TestAPI/bindings/kotlin/src/test/kotlin/com/cricut/testapi/ActorTests.kt
package com.cricut.testapi

import kotlinx.coroutines.test.runTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ActorTests {
    @Test
    fun testLogger() = runTest {
        val logger = Actors.TemperatureLogger.create(label = "log", measurement = 3L)
        logger.update(7L)
        logger.update(1L)
        assertEquals(1L, logger.min())
        assertEquals("gol", logger.backwardsLabel)
    }
}

File: integration-tests/TestAPI/bindings/c-sharp/Cricut.TestAPI.Tests/ActorTests.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;

namespace Cricut.TestAPI.Tests {
    [Collection("root")]
    public class ActorTests {
        [Fact]
        async Task TestActor() {
            var logger = Actors.TemperatureLogger.Create("log", 3);
            await logger.Update(7);
            await logger.Update(1);
            Assert.Equal(1, await logger.Min());
            Assert.Equal("gol", logger.GetBackwardsLabel());
        }
    }
}
